sql/src/main/java/io/crate/analyze/PrivilegesAnalyzer.java

```diff
@@ -42,22 +45,37 @@
  */
 class PrivilegesAnalyzer {
 
-    static PrivilegesAnalyzedStatement analyzeGrant(GrantPrivilege node, @Nullable User user) {
+    private final Schemas schemas;
+
+    PrivilegesAnalyzer(Schemas schemas) {
+        this.schemas = schemas;
+    }
+
+    PrivilegesAnalyzedStatement analyzeGrant(GrantPrivilege node, @Nullable User user) {
         ensureUserManagementEnabled(user);
+        validatePrivilegeIdents(Privilege.Clazz.valueOf(node.clazz()), node.privilegeIdents());
+
         return new PrivilegesAnalyzedStatement(node.userNames(),
-            privilegeTypesToPrivileges(getPrivilegeTypes(node.all(), node.privileges()), user, State.GRANT));
+            privilegeTypesToPrivileges(getPrivilegeTypes(node.all(), node.privileges()), user, State.GRANT, node.privilegeIdents(),
+                Privilege.Clazz.valueOf(node.clazz().toUpperCase(Locale.ENGLISH))));
```

Comment:
afaik the clazz will always be the defined static strings which are always uppercase